app/src/main/java/org/moire/opensudoku/gui/SudokuListSorter.java

```diff
@@ -0,0 +1,46 @@
+package org.moire.opensudoku.gui;
+
+import android.content.Context;
+
+import org.moire.opensudoku.db.SudokuColumns;
+
+public class SudokuListSorter {
+
+    public static final int SORT_BY_CREATED = 0;
+    public static final int SORT_BY_TIME = 1;
+    public static final int SORT_BY_LAST_PLAYED = 2;
+
+    private Context context;
+    private int sortType = SORT_BY_CREATED;
+
+    public SudokuListSorter(Context context) {
+        this(context, SORT_BY_CREATED);
+    }
+
+    public SudokuListSorter(Context context, int sortType) {
+        this.context = context;
+        this.sortType = sortType >= 0 && sortType < 3 ? sortType : SORT_BY_CREATED;
+    }
+
+    public int getSortType() {
+        return sortType;
+    }
+
+    public void setSortType(int sortType) {
+        this.sortType = sortType >= 0 && sortType < 3 ? sortType : SORT_BY_CREATED;;
```

Comment:
I think that in this line there is an extra semicolon

Reply:
Removed. Thank you.